dietpi/dietpi-software

```diff
@@ -11231,7 +11231,8 @@ _EOF_
 			then
 				G_EXEC umask 0037
 				G_EXEC_PRE_FUNC(){ acommand[5]=$GLOBAL_PW; }
-				G_EXEC mosquitto_passwd -c -b /etc/mosquitto/passwd mosquitto "${GLOBAL_PW//?/X}"
+				> /etc/mosquitto/passwd # Pre-create file, required for pre-v1.6.10 (ARMv8 up to Buster): https://github.com/MichaIng/DietPi/issues/4424
+				G_EXEC mosquitto_passwd -b /etc/mosquitto/passwd mosquitto "${GLOBAL_PW//?/X}"
```

Comment:
Somehow it still fails on submitting the password. even if I use a plain pw directly it's failing

```
┌─────────────────────────────────────────────────┤ DietPi-Software ├──────────────────────────────────────────────────┐
│  - Command: mosquitto_passwd -b /etc/mosquitto/passwd mosquitto demo                                                 │
│  - Exit code: 1                                                                                                      │
│  - DietPi version: v7.2.1 (MichaIng/mosquitto) | HW_MODEL: 4 | HW_ARCH: 3 | DISTRO: 5                                │
│  - Image creator: DietPi Core Team                                                                                   │
│  - Pre-image: Raspberry Pi OS Lite (64-bit)                                                                          │
│  - Error log:                                                                                                        │
│ Error: -b argument given but password file, username or password missing.                                            │
```

The only way to get it working is to use `G_EXEC touch` instead of `>`.

Reply:
Strange, `touch` and `>` do exactly the same thing.

Reply:
At least important for me was to us `G_EXEC`

Reply:
Found it. After the `-c` flag has been removed, the wrong command array entry was replaced, replacing the "XXXXX" with the actual password function-internally to avoid printing it to console.

Reply:
yep works now